packages/core/workflows-sdk/src/utils/composer/create-workflow.ts

```diff
@@ -169,22 +180,9 @@ export function createWorkflow<
     return expandedFlow
   }
 
-  let shouldRegisterHookHandler = true
-
-  for (const hook of context.hooks_) {
-    mainFlow[hook] = (fn) => {
-      context.hooksCallback_[hook] ??= []
-
-      if (!shouldRegisterHookHandler) {
-        console.warn(
-          `A hook handler has already been registered for the ${hook} hook. The current handler registration will be skipped.`
-        )
-        return
-      }
-
-      context.hooksCallback_[hook].push(fn)
-      shouldRegisterHookHandler = false
-    }
+  for (const hook of context.hooks_.declared) {
+    mainFlow["hooks"] ??= {}
+    mainFlow["hooks"][hook] = context.hooksCallback_[hook].bind(context)
   }
```

Comment:
nit: we can probably move to `mainFlow.hooks` now that we have added the type, it was temporary 😅

Reply:
It looks like `mainFlow` does not have hooks on it. It is set as type `MainExportedWorkflow`

Reply:
humm, you are right 👍

Reply:
Done